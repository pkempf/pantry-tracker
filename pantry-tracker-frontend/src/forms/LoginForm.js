import React, { useContext } from "react";
import { Redirect } from "react-router-dom";
import Container from "react-bootstrap/Container";
import PantryForm from "./PantryForm";
import UserContext from "../UserContext";

// TEMPORARY
import Button from "react-bootstrap/Button";
import PantryApi from "../api";

// UPON REMOVING THE ABOVE, ALSO REMOVE:
// tempDevLoginHelpers parameter
// devTempLogin() function
// <Button /> block in return statement
// tempDevLoginHelpers variable from App.js
// tempDevLoginHelpers variable from Routes.js

const LoginForm = ({ logInFunction, tempDevLoginHelpers }) => {
  const user = useContext(UserContext);

  if (user.username) {
    return <Redirect to="/" />;
  }

  const devTempLogin = () => {
    PantryApi._setDevTestingToken();
    tempDevLoginHelpers.setToken(
      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZ" +
        "SI6InRlc3R1c2VyIiwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTU5ODE1OTI1OX0." +
        "FtrMwBQwe6Ue-glIFgz_Nf8XxRT2YecFCiSpYL0fCXc"
    );
    tempDevLoginHelpers.setUser({
      username: "testuser",
      firstName: "Test",
      lastName: "User",
      isAdmin: false,
    });
  };

  const formFields = [
    {
      name: "username",
      label: "Username",
      inputType: "text",
      placeholder: "Username",
      initialValue: "",
      required: true,
    },
    {
      name: "password",
      label: "Password",
      inputType: "password",
      placeholder: "Password",
      initialValue: "",
      required: true,
    },
  ];

  return (
    <Container className="LoginForm justify-content-center">
      <PantryForm
        formTitle="Log in"
        fields={formFields}
        submitButtonText="Log in"
        processData={logInFunction}
      />
      <Button className="mt-3" onClick={devTempLogin}>
        DEV LOGIN (TEMPORARY)
      </Button>
    </Container>
  );
};

export default LoginForm;
