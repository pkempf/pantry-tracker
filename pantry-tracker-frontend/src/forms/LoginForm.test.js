import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import LoginForm from "./LoginForm";
import UserContext from "../UserContext";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{}}>
        <LoginForm />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{}}>
        <LoginForm />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
