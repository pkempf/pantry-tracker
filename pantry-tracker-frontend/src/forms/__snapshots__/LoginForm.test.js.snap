// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm justify-content-center container"
  >
    <div
      class="PantryForm justify-content-center row"
    >
      <div
        class="col-md-8 col-sm-10"
      >
        <div
          class="card"
        >
          <h4
            class="card-header"
          >
            Log in
          </h4>
          <div
            class="card-body"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="username"
                >
                  Username
                </label>
                <input
                  class="form-control"
                  id="username"
                  name="username"
                  placeholder="Username"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="password"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  id="password"
                  name="password"
                  placeholder="Password"
                  required=""
                  type="password"
                  value=""
                />
              </div>
              <button
                class="mr-2 btn btn-primary"
                type="submit"
              >
                Log in
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
